db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1180,6 +1185,13 @@ Status DBImpl::CompactRangeInternal(const CompactRangeOptions& options,
                             final_output_level, options, begin, end, exclusive,
                             false /* disable_trivial_move */,
                             std::numeric_limits<uint64_t>::max(), trim_ts);
+  } else if (cfd->ioptions()->compaction_style == kCompactionStyleFIFO) {
+    // Always try to do compaction based on CompactionOptionsFIFO.
+    final_output_level = 0;
+    s = RunManualCompaction(cfd, /*input_level=*/0, final_output_level, options,
+                            begin, end, exclusive,
+                            false /* disable_trivial_move */,
+                            std::numeric_limits<uint64_t>::max(), trim_ts);
```

Comment:
Sorry I'm confused here - I don't see how CompactionOptionsFIFO was considered in this line.

Reply:
It's confusing here. The actual compaction picking happens in `FIFOCompactionPicker::CompactRange()` and there is does not use `begin` or `end` and just does normal fifo compaction. Updated comment for this.